### PR TITLE
Move remote.py into the _cts module

### PR DESCRIPTION
**Pull request by clumens**
Whew, this one got a little out of control.  There's some really weird stuff that gets removed or improved in this PR.

I've run all existing CTS tests and they pass.  I've also examined a diff of before/after logs and they are close but not exact.  Of course, tons of things like pointers and bytes read and PIDs change.  But some other things from the log watcher move places.  I'm not sure if it's a problem or not.  I should run some additional tests and perhaps play around with command line parameters a bit to exercise other code paths.

### CODE REVIEW

File: python/pacemaker/_cts/remote.py [hunk 1]
```diff
@@ -116,29 +116,37 @@ def call_async(self, node, command, delegate=None):
...
+            command     -- The command to run, as a string
+            synchronous -- Should we wait for the command to complete?
+            quiet       -- If True, log the command and its return code but not
+                           its output
+            silent      -- If True, do not log anything - this implies quiet=True
+
+            Returns:
+
+            A tuple of (return code, command output)
+        """
```
Comment (kgaillot) [inline python/pacemaker/_cts/remote.py:185]: Thanks, the return value here has always been confusing.

What do you think about replacing quiet and silent with 0/1/2 verbose or log_level?

**Review by kgaillot — COMMENTED** [review]
Whew! Looks good. Reminds me how much of an overhaul this really needs ...

File: cts/lab/CTSaudits.py [hunk 2]
```diff
@@ -88,7 +92,9 @@ def TestLogging(self):
         if watch_pref == "any": self.CM.log("Writing log with key: %s" % (suffix))
         for node in self.CM.Env["nodes"]:
             cmd = "logger -p %s.info %s %s %s" % (self.CM.Env["SyslogFacility"], prefix, node, suffix)
-            if self.CM.rsh(node, cmd, synchronous=0, silent=True) != 0:
+
+            (rc, _) = self.CM.rsh(node, cmd, synchronous=0, silent=True)
```
[the remark this answers is not in the document]
Comment (clumens) [reply cts/lab/CTSaudits.py:96]: Fixed.

File: cts/lab/watcher.py [hunk 3]
```diff
@@ -92,7 +92,7 @@ def harvest_async(self, delegate=None):
         global log_watcher_bin
         return self.rsh.call_async(self.host,
                                    "%s -t %s -p CTSwatcher: -l 200 -f %s -o %s" % (log_watcher_bin, self.name, self.filename, self.offset),
-                completionDelegate=self)
+                delegate=self)
```
[the remark this answers is not in the document]
Comment (clumens) [reply cts/lab/watcher.py:95]: Fixed.  In general, I've been trying to not change too much in files outside of the one I'm working on, so there's likely a lot more of this kind of stuff to do.

**Comment by clumens** [comment]
> Whew! Looks good. Reminds me how much of an overhaul this really needs ...

My hope is that by modernizing and cleaning up some of the downright weird things happening, it'll be easier to tell what's really going on and then it'll be easier to make more thorough design changes.

**Comment by clumens** [comment]
Updated with new verbose= arg and double checked everywhere it's being called to make sure we're passing the right thing.  Waiting to hear back from QE about qarsh.

File: python/pacemaker/_cts/remote.py [hunk 4]
```diff
@@ -116,36 +116,44 @@ def call_async(self, node, command, delegate=None):
...
+        """ Run the given command on the given remote system.  If you call this class
+            like a function, this is what gets called.  It's approximately the same
+            as a system() call on the remote machine.
+
+            Arguments:
+
+            node        -- The remote machine to run on
+            command     -- The command to run, as a string
+            synchronous -- Should we wait for the command to complete?
+            verbose     -- If 0, do not long anything.  If 1, log the command and its
```
Comment (kgaillot) [inline python/pacemaker/_cts/remote.py:178]: s/long/log/

**Review by kgaillot — COMMENTED** [review]
looks good to me

File: cts/lab/environment.py [hunk 5]
```diff
@@ -310,9 +310,6 @@ def parse_args(self, args):
                 self["DoBSC"] = 1
                 self["scenario"] = "basic-sanity"
 
-            elif args[i] == "--qarsh":
-                RemoteFactory().enable_qarsh()
-
```
[the remark this answers is not in the document]
Comment (clumens) [reply cts/lab/environment.py:315]: QA confirms this is still in use, so I'm dropping this patch.

**Pull request merged**